src/lib/tooltip/tooltip.ts

```diff
@@ -199,11 +203,36 @@ export class MdTooltip implements OnDestroy {
     this._isTooltipVisible() ? this.hide() : this.show();
   }
 
+  _getTooltipId(): string {
+    return this._tooltipInstance ? this._tooltipInstance.id : '';
+  }
+
   /** Returns true if the tooltip is currently visible to the user */
   _isTooltipVisible(): boolean {
     return !!this._tooltipInstance && this._tooltipInstance.isVisible();
   }
 
+  /** Handles the keydown events on the host element. */
+  _handleKeydown(e: KeyboardEvent) {
+    // If the tooltip is visible and the user pressed escape,
+    // intercept the event and hide the tooltip.
+    if (this._tooltipInstance && this._tooltipInstance.isVisible() && e.keyCode === ESCAPE) {
+      e.stopPropagation();
+      this.hide(0);
+    }
+  }
+
+  /** Returns the trigger's aria-describedby attribute. */
+  private _getAriaDescribedby(): string {
+    return this._elementRef.nativeElement.getAttribute('aria-describedby');
```

Comment:
I agree except I think we need would need this anyways since we're checking if the current tooltip matches the aria-describedby and an `Input@` won't capture our setter's interaction.